tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ImmutableMetricName.java

```diff
@@ -60,23 +60,23 @@ private ImmutableMetricName() {}
         }
 
         @CanIgnoreReturnValue
-        public MetricName.Builder putSafeTags(@Safe Map.Entry<String, ? extends String> entry) {
+        public MetricName.Builder putSafeTags(@Safe Map.Entry<@Safe String, @Safe ? extends String> entry) {
             Preconditions.checkNotNull(entry, "entry");
             tagMap = tagMap.withEntry(entry.getKey(), entry.getValue());
             return (MetricName.Builder) this;
         }
 
         @SuppressWarnings("unchecked")
         @CanIgnoreReturnValue
-        public MetricName.Builder safeTags(@Safe Map<String, ? extends String> entries) {
+        public MetricName.Builder safeTags(@Safe Map<@Safe String, @Safe ? extends String> entries) {
             Preconditions.checkNotNull(entries, "entries");
             tagMap = TagMap.of((Map<String, String>) entries);
             return (MetricName.Builder) this;
         }
 
         @SuppressWarnings("unchecked")
         @CanIgnoreReturnValue
-        public MetricName.Builder putAllSafeTags(@Safe Map<String, ? extends String> entries) {
+        public MetricName.Builder putAllSafeTags(@Safe Map<@Safe String, @Safe ? extends String> entries) {
```

Comment:
likely safe to remove the parameter annotation in these cases, but I could go either way